dev/InfoBar/TestUI/InfoBar.cs

```diff
@@ -0,0 +1,428 @@
+            _actionButton = GetTemplateChild<Button>("ActionButton");
+            _myContainer = GetTemplateChild<Border>("Container");
+
+            UpdateButtonsState();
+            UpdateSeverityState();
+            OnIsOpenChanged();
+
+            _alternateCloseButton.Click += new RoutedEventHandler(OnCloseButtonClick);
+            _closeButton.Click += new RoutedEventHandler(OnCloseButtonClick);
+            _actionButton.Click += (s, e) => ActionButtonClick?.Invoke(s, e);
+        }
+
+        private static void OnPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            DependencyProperty property = e.Property;
+            InfoBar infoBar = d as InfoBar;
+            if (property == SeverityProperty)
+            {
+                infoBar.UpdateSeverityState();
+            }
+            else if (property == ActionButtonContentProperty || property == CloseButtonContentProperty || property == ShowCloseButtonProperty)
+            {
+                infoBar.UpdateButtonsState();
+            }
+            else if (property == IsOpenProperty)
+            {
+                infoBar.OnIsOpenChanged();
+            }
+            else if (property == IconSourceProperty)
+            {
+                infoBar.OnIconChanged();
+            }
+        }
+
+        /* Open Properties
+         * 
+         */
+        public bool IsOpen
+        {
+            get { return (bool)GetValue(IsOpenProperty); }
+            set { SetValue(IsOpenProperty, value); }
+        }
+
+        public static readonly DependencyProperty IsOpenProperty =
+            DependencyProperty.Register(nameof(IsOpen), typeof(bool), typeof(InfoBar), new PropertyMetadata(false, OnPropertyChanged));
+
+        public bool ShowCloseButton
+        {
+            get { return (bool)GetValue(ShowCloseButtonProperty); }
+            set { SetValue(ShowCloseButtonProperty, value); }
+        }
+
+        public static readonly DependencyProperty ShowCloseButtonProperty =
+            DependencyProperty.Register(nameof(ShowCloseButton), typeof(bool), typeof(InfoBar), new PropertyMetadata(true, OnPropertyChanged));
+
+        /* Message Title Properties
+         * 
+         */
+        public string Title
+        {
+            get { return (string)GetValue(TitleProperty); }
+            set { SetValue(TitleProperty, value); }
+        }
+
+        public static readonly DependencyProperty TitleProperty =
+            DependencyProperty.Register(nameof(Title), typeof(string), typeof(InfoBar), new PropertyMetadata(""));
+
+        public string Message
+        {
+            get { return (string)GetValue(MessageProperty); }
+            set { SetValue(MessageProperty, value); }
+        }
+
+        public static readonly DependencyProperty MessageProperty =
+            DependencyProperty.Register(nameof(Message), typeof(string), typeof(InfoBar), new PropertyMetadata(""));
+
+        /* Action Button Properties
```

Comment:
Can you merge with master ? This file for example already exists in master, but is showing as newly added.

Reply:
Yes, good idea. I'm currently working on some merge conflicts and will push out some changes to fix it!